src/overlays/actors/ovl_En_Owl/z_en_owl.c

```diff
@@ -31,141 +79,1332 @@ const ActorInit En_Owl_InitVars = {
+    xyzDist = Math3D_Vec3f_DistXYZ(&this->eye, &globalCtx->view.eye) / 45.0f;
+    this->eye.x = globalCtx->view.eye.x;
+    this->eye.y = globalCtx->view.eye.y;
+    this->eye.z = globalCtx->view.eye.z;
+    xyzDist = xyzDist;
+    if (xyzDist > 1.0f) {
+        xyzDist = 1.0f;
+    }
+
+    switch (owlType) {
+        case 7:
+            unkf2 = xyzDist * 2.f;
+            func_800F436C(&D_80ACD62C, 0x20BD, unkf2);
+            if ((0x144 < globalCtx->csCtx.frames) ||
+                ((0x8D < globalCtx->csCtx.frames && (globalCtx->csCtx.frames < 0x10b)))) {
+                func_800F4414(&D_80ACD62C, 0x3926, unkf2);
+            }
+            if (globalCtx->csCtx.frames == 0x55) {
+                func_800F436C(&D_80ACD62C, 0x28BE, unkf2);
+            }
+            break;
+        case 8:
+        case 9:
+            unkf2 = xyzDist * 2.f;
+            func_800F436C(&D_80ACD62C, 0x20BD, unkf2);
+            if ((0x1A3 < globalCtx->csCtx.frames) ||
+                ((0xC1 < globalCtx->csCtx.frames && (globalCtx->csCtx.frames < 0x119)))) {
+                func_800F4414(&D_80ACD62C, 0x3926, unkf2);
+            }
+            if (globalCtx->csCtx.frames == 0xd9) {
+                func_800F436C(&D_80ACD62C, 0x28BE, unkf2);
+            }
+            break;
+    }
+}
+
+void func_80ACB904(EnOwl* this, GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.state != CS_STATE_IDLE && (globalCtx->csCtx.npcActions[7] != NULL)) {
+        if (this->unk_40A != globalCtx->csCtx.npcActions[7]->action) {
+            func_80ACD130(this, globalCtx, 7);
+            func_80ACBAB8(this, globalCtx);
+        }
+        func_80ACD2CC(this, globalCtx);
+    }
+
+    if (this->actionFlags & 0x80) {
+        func_80ACB748(this, globalCtx);
+    }
+}
+
+void func_80ACB994(EnOwl* this, GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.state != CS_STATE_IDLE && (globalCtx->csCtx.npcActions[7] != NULL)) {
+        if (this->unk_40A != globalCtx->csCtx.npcActions[7]->action) {
+            func_80ACD130(this, globalCtx, 7);
+            func_80ACBAB8(this, globalCtx);
+        }
+        func_80ACD4D4(this, globalCtx);
+    }
+
+    if (this->actionFlags & 0x80) {
+        func_80ACB748(this, globalCtx);
+    }
+}
+
+void EnOwl_WaitDefault(EnOwl* this, GlobalContext* globalCtx) {
+    u16 angle;
+    if (globalCtx->csCtx.state != CS_STATE_IDLE && (globalCtx->csCtx.npcActions[7] != NULL)) {
+        if (this->unk_40A != globalCtx->csCtx.npcActions[7]->action) {
+            this->actionFlags |= 4;
+            func_80ACD130(this, globalCtx, 7);
+            func_80ACBAB8(this, globalCtx);
+        } else {
+            this->actor.posRot.rot.z = globalCtx->csCtx.npcActions[7]->urot.y;
+        }
+    }
+
+    if (this->actionFlags & 0x80) {
+        func_80ACB748(this, globalCtx);
+    }
+}
+
+void func_80ACBAB8(EnOwl* this, GlobalContext* globalCtx) {
+    switch (globalCtx->csCtx.npcActions[7]->action - 1) {
+        case CS_STATE_IDLE:
```

Comment:
I could be completely missing the point of this code, but action values shouldn't have anything to do with the cs state enum (which is for `globalCtx->csCtx.state`)